packages/coordinator/src/services/verification/signature-verification.ts

```diff
@@ -64,14 +77,71 @@ function extractSignatureRootFromParentPrescription(
 
 function verifyCertificateValidWhenSigned(parentPrescription: hl7V3.ParentPrescription): boolean {
   const signatureTimeStamp = extractSignatureDateTimeStamp(parentPrescription)
-  const prescriptionCertificate = getX509CertificateFromPrescription(parentPrescription)
+  const prescriptionCertificate = getCertificateFromPrescriptionCrypto(parentPrescription)
   const signatureDate = new Date(convertHL7V3DateTimeToIsoDateTimeString(signatureTimeStamp))
   const certificateStartDate = new Date(prescriptionCertificate.validFrom)
   const certificateEndDate = new Date(prescriptionCertificate.validTo)
   return isDateInRange(signatureDate, certificateStartDate, certificateEndDate)
 }
 
-function getX509CertificateFromPrescription(parentPrescription: hl7V3.ParentPrescription): crypto.X509Certificate {
+async function verifyCertificateRevoked(parentPrescription: hl7V3.ParentPrescription): Promise<boolean> {
+  const prescriptionDate = new Date(convertHL7V3DateTimeToIsoDateTimeString(parentPrescription.effectiveTime))
+  const x509Certificate = getCertificateFromPrescriptionJrsasign(parentPrescription)
+  const serialNumber = x509Certificate.getSerialNumberHex()
+  const distributionPointsURI = x509Certificate.getExtCRLDistributionPointsURI()
+  if (distributionPointsURI) {
+    for (let index = 0; index < distributionPointsURI.length; index++) {
+      const crl = await getRevocationList(distributionPointsURI[index])
```

Comment:
There is a scenario here If the request returns anything other than a 200 we will tell the supplier that the cert isn't revoked.

Reply:
As mentioned in previous conversations, we'll consider it a pass and eventually fallback on checking the full CRL.